tour/01-hello/hello.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.46.4.
.TH HELLO "1" "November 2014" "hello 2.10" "User Commands"
.SH NAME
hello - friendly greeting program
.SH SYNOPSIS
.B hello
[\fI\,OPTION\/\fR]...
.SH DESCRIPTION
Print a friendly, customizable greeting.
.TP
\fB\-h\fR, \fB\-\-help\fR
display this help and exit
.TP
\fB\-v\fR, \fB\-\-version\fR
display version information and exit
.TP
\fB\-t\fR, \fB\-\-traditional\fR
use traditional greeting
.TP
\fB\-g\fR, \fB\-\-greeting\fR=\fI\,TEXT\/\fR
use TEXT as the greeting message
.SH "ADDITIONAL INFORMATION"
See
.UR http://git.savannah.gnu.org/cgit/hello.git/tree/man/hello.x
man/hello.x
.UE
how to add addtional information to a manual page.
.SH "REPORTING BUGS"
Report bugs to: bug\-hello@gnu.org
.br
GNU Hello home page: <http://www.gnu.org/software/hello/>
.br
General help using GNU software: <http://www.gnu.org/gethelp/>
.SH COPYRIGHT
Copyright \(co 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
.br
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
.SH "SEE ALSO"
The full documentation for
.B hello
is maintained as a Texinfo manual.  If the
.B info
and
.B hello
programs are properly installed at your site, the command
.IP
.B info hello
.PP
should give you access to the complete manual.


debian/changelog
snapcraft (2.10) xenial; urgency=medium

  [ Martin Wimpress ]
  * Correct autotools tests to use configflags (#521)

  [ Leo Arias ]
  * Run the integration tests against a local fake server when the user
    password is not in the environment. (#511) (LP: #1585023)
  * Move the login and logout methods to a client. (#518) (LP: #1586504)
  * Improve the config handling. (#519) (LP: #1586511)
  * Fix the one-time password login. (#529) (LP: #1586832)
  * Moved the download to the store client. (#530) (LP: #1586836)
  * Moved the upload to the store client. (#531) (LP: #1586836)
  * Updated the documentation about the icon. (#542) (LP: #1578231)
  * Improve the error message when a part binary is not found. (#541)
    (LP: #1582367)
  * Re-enable the ROS demo for autopackage testing. (#520) (LP: #1588098)
  * Add macaroon support to login, upload and download. (#532) (LP: #1586910)
  * Set the no_proxy environment variable to access the local fake servers.
    (#546) (LP: #1588631)

  [ Stephen Stewart ]
  * nodejs plugin: Support configurable node version (#509) (LP: #1586104)

  [ Kyle Fazzari ]
  * Use correct cross-build packages for ppc64le. (#539) (LP: #1570944)

  [ Sergio Schvezov ]
  * Support zip files as source (#523) (LP: #1577062)
  * A nicer error message for incorrect stage-packages (#524) (LP: #1568131)
  * Support the assumes keyword (#525) (LP: #1586429)
  * Improve the template for snapcraft init (#528) (LP: #1575581)
  * Filter out *.snap from sourcedir (#535) (LP: #1575628)
  * Support setting a gopath for a go project from vcs (#538) (LP: #1583426)
  * Add a ticker for snapping (#540) (LP: #1582955)
  * Rename strip to prime (#543) (LP: #1582515)

  [ Didier Roche ]
  * Wrap plugin list output content (#534) (LP: #1587057)
  * Add snapcraft examples to scaffold getting started tour (#513)
    (LP: #1586137)

  [ Joe Talbott ]
  * Add support for parsing the parts wiki (#545) (LP: #1587583)

 -- Sergio Schvezov <sergio.schvezov@ubuntu.com>  Fri, 03 Jun 2016 13:37:58 -0300

snapcraft (2.9) xenial; urgency=medium

  [ Leo Arias ]
  * autopkgtests: run the install examples tests in classic. (#481)
    (LP: #1572764)

  [ Matteo Bertini ]
  * Fix typo in description of the python3 example. (#504)

  [ Jamie Bennett ]
  * Documentation: Use plugs instead of caps. (#507)

  [ Chris Wayne ]
  * Add in bash completion. (#453) (LP: #1570506)

  [ Sergio Schvezov ]
  * Fail validation if plugs or slots are declared at the part level (#514)
    (LP: #1581166)

  [ Kyle Fazzari ]
  * Make pull and build steps dirty if target arch changes. (#450)
    (LP: #1564192)
  * Add support for the confinement property. (#501) (LP: #1580819)
  * Add support for the epoch property. (#502) (LP: #1581113)

 -- Sergio Schvezov <sergio.schvezov@ubuntu.com>  Tue, 24 May 2016 23:32:11 -0300

snapcraft (2.8.8) xenial; urgency=medium

  * Create the 'partial' directory for apt. (#499) (LP: #1578007)

 -- Sergio Schvezov <sergio.schvezov@ubuntu.com>  Wed, 04 May 2016 14:22:58 -0300

snapcraft (2.8.7) xenial; urgency=medium

  * Don't fail if there is no system library list (#496) (LP: #1577750)

 -- Sergio Schvezov <sergiusens@gmail.com>  Tue, 03 May 2016 10:58:57 -0300

snapcraft (2.8.6) xenial; urgency=medium

  [ Leo Arias ]
  * Import mock from unittest. (#492) (LP: #1576998)

  [ Sergio Schvezov ]
  * Remove missing replacement for unittest.mock (#494) (LP: #1576998)

 -- Sergio Schvezov <sergio.schvezov@ubuntu.com>  Sat, 30 Apr 2016 18:44:44 -0300

snapcraft (2.8.5) xenial; urgency=medium

  [ Leo Arias ]
  * autopkgtests: run the tests using the installed package (#464)
    (LP: #1570992)
  * Remove --allow-unauthenticated from examples tests (#482) (LP: #1573211)
  * Update the assertion of the example install (#483) (LP: #1573243)
  * Examples tests: Update the path to the snaps binaries (#484) (LP: #1573349)
  * Examples tests: use systemctl instead of the removed snap service (#485)
    (LP: #1573697)
  * Update the mosquitto example SNAP_USER_DATA path. (#486) (LP: #1574857)
  * Update the busybox test to use the snap data path. (LP: #1574901)
  * Remove the integration tests coverage. (#488) (LP: #1575383)

  [ Sergio Schvezov ]
  * Don't clean target before extracting npm (#489) (LP: #1575876)
  * Don't delete the nodejs download on build. (#490) (LP: #1575882)

  [ Vincent Ladeuil ]
  * Make upload more robust by ignoring spurious errors while polling the
    scan status. (#480) (LP: #1572963)

  [ Kyle Fazzari ]
  * docs/get-started.md: Stop discussing snappy-tools. (#454) (LP: #1568113)

 -- Sergio Schvezov <sergio.schvezov@ubuntu.com>  Fri, 29 Apr 2016 14:59:47 -0300

snapcraft (2.8.4) xenial; urgency=medium

  [ Sergio Schvezov ]
  * Use series 16 by default instead of rolling-core (#477) (LP: #1572602)
  * kernel plugin: iterate over modules to add to initrd (#476) (LP: #1572118)

  [ Kyle Fazzari ]
  * Follow symlinks while hard-linking migrated files. (#478) (LP: #1572664)

 -- Sergio Schvezov <sergio.schvezov@ubuntu.com>  Wed, 20 Apr 2016 19:15:32 -0300

snapcraft (2.8.3) xenial; urgency=medium

  [ Kyle Fazzari ]
  * docs/intro.md: Fix broken developer reference link. (#456) (LP: #1558296)
  * Nil plugin: Make purpose more clear in documentation. (#455) (LP: #1549691)
  * Use colors for logging. (#474) (LP: #1572186)

  [ Leo Arias ]
  * Update the snapcraft-coverage script to use the internal dirs (#471)
    (LP: #1571696)

  [ Sergio Schvezov ]
  * Fix shell quoting (#473) (LP: #1572129)

 -- Sergio Schvezov <sergio.schvezov@ubuntu.com>  Tue, 19 Apr 2016 17:59:45 -0300

snapcraft (2.8.2) xenial; urgency=medium

  * Don't expose broken or fragile APIs to plugins (LP: #1571264)
  * catkin plugin: use ProjectOptions for repo setup (#468) (LP: #1571446)
  * qml plugin: remove broken plugin (#469) (LP: #1551343)

 -- Sergio Schvezov <sergio.schvezov@ubuntu.com>  Mon, 18 Apr 2016 10:06:15 -0300

snapcraft (2.8.1) xenial; urgency=medium

  [ Simon Busch ]

tour/01-hello/src/hello.c
/* hello.c -- print a greeting message and exit.

   Copyright 1992, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2005,
   2006, 2007, 2008, 2010, 2011, 2013, 2014 Free Software Foundation,
   Inc.

   This program is free software: you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation, either version 3 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program.  If not, see <http://www.gnu.org/licenses/>.  */

#include <config.h>
#include "system.h"
#include "errno.h"
#include "error.h"
#include "progname.h"
#include "xalloc.h"

static const struct option longopts[] = {
  {"greeting", required_argument, NULL, 'g'},
  {"help", no_argument, NULL, 'h'},
  {"traditional", no_argument, NULL, 't'},
  {"version", no_argument, NULL, 'v'},
  {NULL, 0, NULL, 0}
};

/* Forward declarations.  */
static void print_help (void);
static void print_version (void);

int
main (int argc, char *argv[])
{
  int optc;
  int lose = 0;
  const char *greeting_msg;
  wchar_t *mb_greeting;
  size_t len;

  set_program_name (argv[0]);

  /* Set locale via LC_ALL.  */
  setlocale (LC_ALL, "");

#if ENABLE_NLS
  /* Set the text message domain.  */
  bindtextdomain (PACKAGE, LOCALEDIR);
  textdomain (PACKAGE);
#endif

  /* Having initialized gettext, get the default message. */
  greeting_msg = _("Hello, world!");

  /* Even exiting has subtleties.  On exit, if any writes failed, change
     the exit status.  The /dev/full device on GNU/Linux can be used for
     testing; for instance, hello >/dev/full should exit unsuccessfully.
     This is implemented in the Gnulib module "closeout".  */
  atexit (close_stdout);

  while ((optc = getopt_long (argc, argv, "g:htv", longopts, NULL)) != -1)
    switch (optc)
      {
	/* --help and --version exit immediately, per GNU coding standards.  */
      case 'v':
	print_version ();
	exit (EXIT_SUCCESS);
	break;
      case 'g':
	greeting_msg = optarg;
	break;
      case 'h':
	print_help ();
	exit (EXIT_SUCCESS);
	break;
      case 't':
	greeting_msg = _("hello, world");
	break;
      default:
	lose = 1;
	break;
      }

  if (lose || optind < argc)
    {
      /* Print error message and exit.  */
      error (0, 0, "%s: %s", _("extra operand"), argv[optind]);
      print_help ();
    }

  len = mbsrtowcs(NULL, &greeting_msg, 0, NULL);
  if (len == (size_t)-1)
    error (EXIT_FAILURE, errno, _("conversion to a multibyte string failed"));
  mb_greeting = xmalloc((len + 1) * sizeof(wchar_t));
  mbsrtowcs(mb_greeting, &greeting_msg, len + 1, NULL);

  /* Print greeting message and exit. */
  wprintf (L"%ls\n", mb_greeting);
  free(mb_greeting);

  exit (EXIT_SUCCESS);
}


/* Print help info.  This long message is split into
   several pieces to help translators be able to align different
   blocks and identify the various pieces.  */

static void
print_help (void)
{
  /* TRANSLATORS: --help output 1 (synopsis)
     no-wrap */
  printf (_("\
Usage: %s [OPTION]...\n"), program_name);

  /* TRANSLATORS: --help output 2 (brief description)
     no-wrap */
  fputs (_("\
Print a friendly, customizable greeting.\n"), stdout);

  puts ("");
  /* TRANSLATORS: --help output 3: options 1/2
     no-wrap */
  fputs (_("\
  -h, --help          display this help and exit\n\
  -v, --version       display version information and exit\n"), stdout);

  puts ("");
  /* TRANSLATORS: --help output 4: options 2/2
     no-wrap */
  fputs (_("\
  -t, --traditional       use traditional greeting\n\
  -g, --greeting=TEXT     use TEXT as the greeting message\n"), stdout);

  printf ("\n");
  /* TRANSLATORS: --help output 5+ (reports)
     TRANSLATORS: the placeholder indicates the bug-reporting address
     for this application.  Please add _another line_ with the
     address for translation bugs.
     no-wrap */
  printf (_("\
Report bugs to: %s\n"), PACKAGE_BUGREPORT);
#ifdef PACKAGE_PACKAGER_BUG_REPORTS
  printf (_("Report %s bugs to: %s\n"), PACKAGE_PACKAGER,
	  PACKAGE_PACKAGER_BUG_REPORTS);
#endif
#ifdef PACKAGE_URL
  printf (_("%s home page: <%s>\n"), PACKAGE_NAME, PACKAGE_URL);
#else
  printf (_("%s home page: <http://www.gnu.org/software/%s/>\n"),
	  PACKAGE_NAME, PACKAGE);
#endif
  fputs (_("General help using GNU software: <http://www.gnu.org/gethelp/>\n"),
	 stdout);
}



/* Print version and copyright information.  */

static void
print_version (void)
{
  printf ("%s (%s) %s\n", PACKAGE, PACKAGE_NAME, VERSION);
  /* xgettext: no-wrap */
  puts ("");

  /* It is important to separate the year from the rest of the message,
     as done here, to avoid having to retranslate the message when a new
     year comes around.  */
  printf (_("\
Copyright (C) %d Free Software Foundation, Inc.\n\
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>\n\
This is free software: you are free to change and redistribute it.\n\
There is NO WARRANTY, to the extent permitted by law.\n"), COPYRIGHT_YEAR);
}
